File: pad_icon.dart
import 'dart:io';
import 'package:image/image.dart' as img;

void main() {
  final candidates = [
    'assets/icon/app_icon_v3.png',
    'assets/icon/app_icon_v2.png',
    'assets/icon/app_icon.png'
  ];
  
  img.Image? image;
  String? selectedPath;
  
  for (final path in candidates) {
    print('Trying to load $path...');
    final file = File(path);
    if (!file.existsSync()) continue;
    image = img.decodeImage(file.readAsBytesSync());
    if (image != null) {
      selectedPath = path;
      print('Successfully decoded $path');
      break;
    }
  }
  
  if (image == null) {
    print('Failed to decode any icon image.');
    return;
  }
  
  final width = image.width;
  final height = image.height;
  print('Original size: $width x $height');
  
  // Scale down to 58% to fit perfectly inside adaptive icon safe zone
  final newW = (width * 0.58).toInt();
  final newH = (height * 0.58).toInt();
  print('Resizing logo to: $newW x $newH');
  
  final resized = img.copyResize(image, width: newW, height: newH, interpolation: img.Interpolation.cubic);
  
  // Create transparent canvas
  final canvas = img.Image(width: width, height: height, numChannels: 4);
  img.fill(canvas, color: img.ColorRgba8(0, 0, 0, 0));
  
  // Paste resized image in the center
  final offsetX = (width - newW) ~/ 2;
  final offsetY = (height - newH) ~/ 2;
  
  img.compositeImage(canvas, resized, dstX: offsetX, dstY: offsetY);
  
  final outPath = 'assets/icon/app_icon_adaptive_foreground.png';
  File(outPath).writeAsBytesSync(img.encodePng(canvas));
  print('Successfully generated adaptive icon foreground at: $outPath');
}

File: pad_icon.py
import sys
from PIL import Image

def pad_image():
    try:
        img_path = 'assets/icon/app_icon_v3.png'
        out_path = 'assets/icon/app_icon_adaptive_foreground.png'
        
        img = Image.open(img_path).convert("RGBA")
        width, height = img.size
        
        # Scale down to 62% to fit perfectly within Android's adaptive icon safe zone (72/108 = 66%, 62% gives extra safety margin)
        new_w = int(width * 0.62)
        new_h = int(height * 0.62)
        
        resized_img = img.resize((new_w, new_h), Image.Resampling.LANCZOS)
        
        # Create new transparent image of original size
        new_img = Image.new("RGBA", (width, height), (0, 0, 0, 0))
        
        # Paste resized image in the center
        offset_x = (width - new_w) // 2
        offset_y = (height - new_h) // 2
        new_img.paste(resized_img, (offset_x, offset_y), resized_img)
        
        new_img.save(out_path, "PNG")
        print("Successfully created padded adaptive icon foreground at:", out_path)
    except Exception as e:
        print("Error padding image:", e)
        sys.exit(1)

if __name__ == '__main__':
    pad_image()
